src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,119 @@
 package co.inventorsoft.academy.collections.model;
 
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Set;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+    private HashSet<T> range;
+
+    public Range(){
+        range = new HashSet<T>();
+    }
+
+    static <T> Range<T> of(T start, T end) {
+        Range<T> rT = new Range<>();
+        if (start.equals(end)) {
+            return rT;
+        }
+
+        if(start instanceof Integer && end instanceof Integer){
```

Comment:
Don't do that, create for Integer and Float separate methods as it was.
But, instead of implementing all the logic inside those methods (for Float and Integer) just call method of(T start, T end, Function<T, T> function)

Reply:
let me give you an idea, inside of method of(Integer, Integer) call Range.of(start, end, new Function<Integer, Integer>() {

    @Override
    public Integer apply(Integer o) {
        return o+1;
    }
});